app/assets/javascripts/discourse-common/addon/lib/icon-library.js

```diff
@@ -132,9 +812,34 @@ export function isExistingIconId(id) {
   return _iconList?.includes(id);
 }
 
-function warnIfMissing(id) {
-  if (warnMissingIcons && isDevelopment() && !isExistingIconId(id)) {
-    console.warn(`The icon "${id}" is missing from the SVG subset.`); // eslint-disable-line no-console
+function handleDeprecatedIcon(id) {
+  if (DEPRECATIONS.hasOwnProperty(id)) {
+    const newId = DEPRECATIONS[id];
+
+    // eslint-disable-next-line no-console
+    console.warn(
+      `The icon name "${id}" has been updated to "${newId}". Old names will be removed in Q2 2025.`
+    );
```

Comment:
We should use our `deprecated()` function. That way, it'll throw errors in core CI, print useful warnings in plugin/theme CI,  and will allow us to keep track of any pending issues in production via the discourse-deprecation-collector